### DISCUSSION TITLE
RC DB migration of encrypted settings causing fatal error?

### DISCUSSION THREAD

**Discussion by pnmcosta** [pocketbase/pocketbase#5623] [Q&A]
Hi Gani,

First of all great work yet again, with the refactoring taken for 0.23 it is looking great and I really appreciate the changes, thank you.

I'm testing the data migration from v0.22.21 to v0.23.0-rc3, my production database settings are encrypted as per the recommended approach on docs.

I've taken a copy of the prod database and am running it against the new RC version with `go run main.go serve --encryptionEnv=APP_ENCRYPTION_KEY --dev` and always yields the following error (trace tailed):

```
...
[0.00ms] SELECT `_collections`.* FROM `_collections` ORDER BY `created` ASC
[0.00ms] INSERT INTO `_migrations` (`applied`, `file`) VALUES (1728310695968133, '1717233556_v0.23_migrate.go')
[0.00ms] SELECT `_params`.* FROM `_params` WHERE `id`='settings'
2024/10/07 15:18:15 transaction afterFunc errors: illegal base64 data at input byte 4
exit status 1
```

Please note the following:

- Doing the same on v0.22.21 yields no errors and the app starts successfully, I triple checked this 😅
- https://github.com/pnmcosta/pb-rc-poc for the minimum test setup I'm using (but any as framework setup should reproduce it)
- There are no new or existing migrations on the test setup, I'm thoroughly following the upgrade guide but wanted to have the DB migrated before all my app's code is, AFAIK this process should work OTB, correct?
- I am using the same encryption/decryption method (and key) on a custom collection's JSON field, could it be related to that instead? I don't think so because I assume it would only be observable when actually querying those collection records, but you would probably know better than I since I haven't dug deep into the internal migration.

Let me know if there's anything else I should be looking at specifically to help you troubleshoot this.

**Comment by pnmcosta** [comment]
Bit more info, on my dev env I do not leverage encryption at all, I created a brand new data.db file (by removing the old one) and let all my migrations run with v0.22.21.

When using this db version on the v0.23.0-rc3 it always yields the following and hangs (does not exit 1):

```
...
[0.00ms] INSERT INTO `_migrations` (`applied`, `file`) VALUES (1728313941639479, '1717233556_v0.23_migrate.go')
[0.00ms] SELECT `_params`.* FROM `_params` WHERE `id`='settings'
2024/10/07 16:12:21 transaction afterFunc errors: invalid settings db data or missing encryption key ""
```

Even though I'm running it only with `go run main.go serve --dev` (without specifying a `--encryptionEnv` param).

I've included below a formatted copy paste of this database `_params.settings` key value, which was not encrypted as expected. I have replaced all secrets and tokens with `XXXX` and the only one using any special characters is the `smtp.password` value that has a JWT token like format (with dots).

```
{
    "meta": {
        "appName": "my-app",
        "appUrl": "https://my.app",
        "hideControls": true,
        "senderName": "Support",
        "senderAddress": "support@example.com",
        "verificationTemplate": {
            "body": "\u003cp\u003eHello,\u003c/p\u003e\n\u003cp\u003eThank you for joining us at {APP_NAME}.\u003c/p\u003e\n\u003cp\u003eClick on the button below to verify your email address.\u003c/p\u003e\n\u003cp\u003e\n  \u003ca class=\"btn\" href=\"{ACTION_URL}\" target=\"_blank\" rel=\"noopener\"\u003eVerify\u003c/a\u003e\n\u003c/p\u003e\n\u003cp\u003e\n  Thanks,\u003cbr/\u003e\n  {APP_NAME} team\n\u003c/p\u003e",
            "subject": "Verify your {APP_NAME} email",
            "actionUrl": "{APP_URL}/_/#/auth/confirm-verification/{TOKEN}",
            "hidden": false
        },
        "resetPasswordTemplate": {
            "body": "\u003cp\u003eHello,\u003c/p\u003e\n\u003cp\u003eClick on the button below to reset your password.\u003c/p\u003e\n\u003cp\u003e\n  \u003ca class=\"btn\" href=\"{ACTION_URL}\" target=\"_blank\" rel=\"noopener\"\u003eReset password\u003c/a\u003e\n\u003c/p\u003e\n\u003cp\u003e\u003ci\u003eIf you didn't ask to reset your password, you can ignore this email.\u003c/i\u003e\u003c/p\u003e\n\u003cp\u003e\n  Thanks,\u003cbr/\u003e\n  {APP_NAME} team\n\u003c/p\u003e",
            "subject": "Reset your {APP_NAME} password",
            "actionUrl": "{APP_URL}/_/#/auth/confirm-password-reset/{TOKEN}",
            "hidden": false
        },
        "confirmEmailChangeTemplate": {
            "body": "\u003cp\u003eHello,\u003c/p\u003e\n\u003cp\u003eClick on the button below to confirm your new email address.\u003c/p\u003e\n\u003cp\u003e\n  \u003ca class=\"btn\" href=\"{ACTION_URL}\" target=\"_blank\" rel=\"noopener\"\u003eConfirm new email\u003c/a\u003e\n\u003c/p\u003e\n\u003cp\u003e\u003ci\u003eIf you didn't ask to change your email address, you can ignore this email.\u003c/i\u003e\u003c/p\u003e\n\u003cp\u003e\n  Thanks,\u003cbr/\u003e\n  {APP_NAME} team\n\u003c/p\u003e",
            "subject": "Confirm your {APP_NAME} new email address",
            "actionUrl": "{APP_URL}/_/#/auth/confirm-email-change/{TOKEN}",
            "hidden": false
        }
    },
    "logs": {
        "maxDays": 7,
        "minLevel": 0,
        "logIp": true
    },
    "smtp": {
        "enabled": true,
        "host": "smtp.sendgrid.net",
        "port": 587,
        "username": "XXXX",
        "password": "XXXX",
        "authMethod": "",
        "tls": false,
        "localName": ""
    },
    "s3": {
        "enabled": false,
        "bucket": "",
        "region": "",
        "endpoint": "",
        "accessKey": "",
        "secret": "",
        "forcePathStyle": false
    },
    "backups": {
        "cron": "",
        "cronMaxKeep": 3,
        "s3": {
            "enabled": false,
            "bucket": "",
            "region": "",
            "endpoint": "",
            "accessKey": "",
            "secret": "",
            "forcePathStyle": false
        }
    },
    "adminAuthToken": {
        "secret": "XXXX",
        "duration": 1209600
    },
    "adminPasswordResetToken": {
        "secret": "XXXX",
        "duration": 1800
    },
    "adminFileToken": {
        "secret": "XXXX",
        "duration": 120
    },
    "recordAuthToken": {
        "secret": "XXXX",
        "duration": 1209600
    },
    "recordPasswordResetToken": {
        "secret": "XXXX",
        "duration": 1800
    },
    "recordEmailChangeToken": {
        "secret": "XXXX",
        "duration": 1800
    },
    "recordVerificationToken": {
        "secret": "XXXX",
        "duration": 604800
    },
    "recordFileToken": {
        "secret": "XXXX",
        "duration": 120
    },
    "emailAuth": {
        "enabled": false,
        "exceptDomains": null,
        "onlyDomains": null,
        "minPasswordLength": 0
    },
    "googleAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "facebookAuth": {
        "enabled": true,
        "clientId": "XXXX",
        "clientSecret": "XXXX",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "githubAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "gitlabAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "discordAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "twitterAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "microsoftAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "spotifyAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "kakaoAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "twitchAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "stravaAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "giteeAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "livechatAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "giteaAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "oidcAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "oidc2Auth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "oidc3Auth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "appleAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "instagramAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "vkAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "yandexAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "patreonAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "mailcowAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "bitbucketAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    },
    "planningcenterAuth": {
        "enabled": false,
        "clientId": "",
        "clientSecret": "",
        "authUrl": "",
        "tokenUrl": "",
        "userApiUrl": "",
        "displayName": "",
        "pkce": null
    }
}
```

**Comment by ganigeorgiev** [comment]
I'm confused and not sure how to reproduce it. The steps I've attempted:

```
# 0. clone the repo
git clone https://github.com/pnmcosta/pb-rc-poc.git

# 1. Navigate to the cloned repo and extract a PocketBase v0.22.21 binary inside it
cd pb-rc-poc
...

# 2. Set encryption key
export PB_ENCRYPTION_KEY="xMTbb6ruiwm4gg7Pal2gt9HKKR03sWeH"

# 3. Start the v0.22.21 binary
./pocketbase serve --encryptionEnv=PB_ENCRYPTION_KEY

# 4. Initialize the database and do some settings changes from the UI...

# 5. stop the v0.22.21 process and run the poc on top of the just created pb_data
go run main.go serve --encryptionEnv=PB_ENCRYPTION_KEY
```

> I am using the same encryption/decryption method (and key) on a custom collection's JSON field, could it be related to that instead? 

I don't understand what do you mean by this.

**Reply by pnmcosta** [reply]
Oh wow, maybe it's my env go cache that is messed up. I'm going to `go clean -r -cache` and re run everything again, sorry about that if that's the problem.

However, how were you able to extract a PocketBase v0.22.21 binary from the PoC as it is targeting the RC on the go.mod? See [here](https://github.com/pnmcosta/pb-rc-poc/blob/main/go.mod#L7) ?

**Reply by ganigeorgiev** [reply]
The binary is extracted manually from the PocketBase v0.22.21 archive downloaded from the GitHub releases. I didn't run `go build`.

**Reply by ganigeorgiev** [reply]
If you can work with a `develop` branch fork (_you'll have to use the `replace` directive in go.mod to point to the local version_), could you try in the place of the `// decrypt` block in  https://github.com/pocketbase/pocketbase/blob/9b5aaba34f9fb99a0035e881e3571bd9574f8c26/core/settings_query.go#L72-L76:

```go
strValue := string(param.Value)
if len(strValue) > 1 && strValue[0] == '"' && strValue[len(strValue)-1] == '"' {
    strValue = strValue[1 : len(strValue)-1]
}

// decrypt
decrypted, decryptErr := security.Decrypt(strValue, encryptionKey)
if decryptErr != nil {
    return decryptErr
}
```

The above might be needed because in earlier PocketBase version we were storing the params value as encoded json string and this may not be reproduced when a more new `pb_data` is used.

I'll try a little bit later this evening with an older version to try to recreate it.

**Reply by pnmcosta** [reply]
Ah I see, I misread your note, you copied the binary into the repo clone location 🤦

**Reply by pnmcosta** [reply]
> If you can work with a develop branch fork (you'll have to use the replace directive in go.mod to point to the local version)

Yup can do, and will do. I'm just re-testing everything after clearing the go cache, it could have just been a red-hering from my side 🥲

**Reply by pnmcosta** [reply]
I've retested and still encountering the same issues. The earliest version I used of PB was v0.14.3

I'm now going to try the changes you suggested locally.

**Reply by pnmcosta** [reply]
That didn't work. But I have discovered that running a second time the app starts and the db's are migrated correctly both the encrypted version and the one that is not.

**Reply by ganigeorgiev** [reply]
I'm not really sure in this case what is causing it. Based on the initial error message it is possible to be a padding issue since now looking at the godocs I noticed that there are 2 slightly different encoder fields - `base64.StdEncoding` and `base64.RawStdEncoding` (_internally both for encrypt/decrypt `StdEncoding` is used and it expected to produce compatible outputs..._). I'll have to experiment later to see if I can reproduce the error (_if that is where it is breaking in the first place_).

**Comment by pnmcosta** [comment]
I've added 
```
fmt.Println(fmt.Errorf("plainDecodeErr: %s", plainDecodeErr))
fmt.Printf("value: %s\n", param.Value)
```

above https://github.com/pocketbase/pocketbase/blob/9b5aaba34f9fb99a0035e881e3571bd9574f8c26/core/settings_query.go#L65

and yields:

```
plainDecodeErr: invalid character '-' after top-level value
value: 2024-10-07 16:31:34.215Z
```

I don't think the query is fetching the correct column value after the migration runs. I'm going deeper 😅

**Reply by ganigeorgiev** [reply]
This is expected to fail if the settings were previously encrypted.

**Reply by ganigeorgiev** [reply]
Can you confirm that you can still reproduce it if you run it on top of new PocketBase v0.22.21 `pb_data` as shown in https://github.com/pocketbase/pocketbase/discussions/5623#discussioncomment-10869605?

**Reply by pnmcosta** [reply]
No, I cannot reproduce it on v0.22.21 setup at all. But I'm going to follow your instructions exactly.

**Reply by pnmcosta** [reply]
notice one thing tho, I've added some debug prints to ReloadSettings method as such

```
func (app *BaseApp) ReloadSettings() error {
	param := &Param{}
	err := app.ModelQuery(param).Model(paramsKeySettings, param)
	if err != nil && !errors.Is(err, sql.ErrNoRows) {
		return err
	}

	fmt.Printf("param on reload: %+v\n", param)

	// no settings were previously stored -> save
	// (ReloadSettings() will be invoked again by a system hook after successful save)
	if param.Id == "" {
		// force insert in case the param entry was deleted manually after application start
		app.Settings().MarkAsNew()
		return app.Save(app.Settings())
	}

	event := new(SettingsReloadEvent)
	event.App = app

	return app.OnSettingsReload().Trigger(event, func(e *SettingsReloadEvent) error {
		fmt.Printf("param on trigger: %+v\n", param)
		return e.App.Settings().loadParam(e.App, param)
	})
}
```

And those yielded on the first run (migration ran):

```
[0.00ms] SELECT `_params`.* FROM `_params` WHERE `id`='settings'
param on reload: &{BaseModel:{lastSavedPK:settings Id:settings} Created:2024-10-07 16:36:20.468Z Updated: Value:2024-10-07 16:36:20.468Z}
param on trigger: &{BaseModel:{lastSavedPK:settings Id:settings} Created:2024-10-07 16:36:20.468Z Updated: Value:2024-10-07 16:36:20.468Z}
plainDecodeErr: invalid character '-' after top-level value
value: 2024-10-07 16:36:20.468Z
2024/10/07 17:36:20 transaction afterFunc errors: invalid settings db data or missing encryption key ""
exit status 1
```

But on the second run, then the value as filled with the actual settings instead of the created/updated date.

**Reply by ganigeorgiev** [reply]
Sorry but this really confusing and I don't know how to help you.

Try to to list the exact steps that you've taken to reproduce it, otherwise we are just going in circles.

**Reply by pnmcosta** [reply]
> This is expected to fail if the settings were previously encrypted.

Is correct, but because the value fetched is actually a date, it's erroring then and after on the decrypt attempt.

**Reply by pnmcosta** [reply]
> Try to to list the exact steps that you've taken to reproduce it

I'm going to do exactly what you did on https://github.com/pocketbase/pocketbase/discussions/5623#discussioncomment-10869605

Except that in between 1. and 2. I'm going to copy my production v0.22.21 of the database, cause effectively that's what's causing issues and not a newly synthed DB from v0.22.21

**Reply by pnmcosta** [reply]
Yup same problem. This is what I did:

```
# 1. Navigate to the cloned repo and extract a PocketBase v0.22.21 binary inside it
cd pb-rc-poc
...

# 1.5 Copy the v0.22.21 production db that is encrypted

# 2. Set encryption key (xxx is not actual value)
export PB_ENCRYPTION_KEY="xxx"

# 3. Start the v0.22.21 binary
./pocketbase serve --encryptionEnv=PB_ENCRYPTION_KEY

# 4. Initialize the database and do some settings changes from the UI...

# 5. stop the v0.22.21 process and run the poc on top of the just created pb_data
go run main.go serve --encryptionEnv=PB_ENCRYPTION_KEY
```

No problems with step 3. when running step 5. got the issue I've reported before here https://github.com/pocketbase/pocketbase/discussions/5623#discussioncomment-10870619

**Comment by ganigeorgiev** [comment]
I was able to reproduce it following the steps in https://github.com/pocketbase/pocketbase/discussions/5623#discussioncomment-10869605, but only with `CGO_ENABLED=1`.

I'm not sure why exactly this is failing with the CGO driver but will investigate it later this evening or tomorrow in the morning.

**Reply by pnmcosta** [reply]
ah yes, my `go env` does have CGO_ENABLED='1' hence why my `go run main.go` were failing too. I'm glad it's caught 🙌

**Reply by pnmcosta** [reply]
on a side note, I took the early decision to use CGO on performance concerns, that I may have misinterpreted, and in hindsight I now kind of regret, and may use this upgrade do disable it

in your experience do you foresee any issues with disabling it now? I will be migrating the existing prod database that has always had CGO enabled, and I'm hosting it on fly.io, which should be straight forward to disable (by removing the build image dependency on it).

**Reply by ganigeorgiev** [reply]
I've opened an issue in upstream repo - https://github.com/mattn/go-sqlite3/issues/1284.

I think it is a bug with the CGO driver but it is a little difficult to reproduce directly when there are several layers (PocketBase, dbx, database/sql, driver) and it also requires at least 2 connections to be active (aka. it could be affected by the idle connections timeout).

This is not PocketBase v0.23.0 specific and old versions are also buggy.

The main culprit in our case is the driver `rows.Columns()` method which returns an old state if the connection that invokes it is opened BEFORE the schema change applied by another connection (e.g. 1 read and 1 write).

This is the reason why when you posted in https://github.com/pocketbase/pocketbase/discussions/5623#discussioncomment-10870619, `Param.Value` is populated incorrectly with some of timestamp column value because the internal columns counter is out of sync - in v0.23.0 the `key` column is deleted and from 5 we now have 4 columns (`id`, `value`, `created`, `updated`).

This also explains why when you run the app again the error disapear because the new db connections are opened AFTER the already applied schema change.

Strangely you can force the columns sync between the connections if you add `app.DB().NewQuery("PRAGMA integrity_check").Execute()` at the beginning of `ReloadSettings()` (_optimize/quick_check and maybe other PRAGMAs also seems to work_).

For now my recommendation is to use the pure Go driver, aka. build with `CGO_ENABLED=0`.

Both drivers should be able to read and modify the same db file so it should be safe even if your existing db was created with the CGO driver.

**Reply by pnmcosta** [reply]
Thank you Gani for your diligence on getting this traced 🙌

From my part, I'll make sure to share my `go env` going forth when reporting any future issues, but I am going to go with `CGO_ENABLED=0` going forth as advised.

If calling `app.DB().NewQuery("PRAGMA integrity_check").Execute()` seems to fix the columns cache on the driver, couldn't a temporary patch be to do it as the last step of the migration?

BTW, I also think this might somewhat be impacting the the `logs.db` to `auxiliary.db` rename, I observed some errors as I was conducting these tests but didn't want to confuse it further.

**Reply by ganigeorgiev** [reply]
> If calling app.DB().NewQuery("PRAGMA integrity_check").Execute() seems to fix the columns cache on the driver, couldn't a temporary patch be to do it as the last step of the migration?

Unfortunately No because it needs to be ran against each live opened connection maintained by the `sql.DB` pool in order to be reliable and we don't have direct access to the pool connections (they are dynamic and affected by various factors like the max opened connections limit, max open idle connections, max connection lifetime, max idle connection lifetime, etc.).

You can create or extract an existing connection with the [`Conn()` method](https://pkg.go.dev/database/sql#DB.Conn), but the pool picks them for you.

Regarding the errors related to the `auxiliary.db` it could be related to https://github.com/pocketbase/pocketbase/discussions/5618 (it will be resolved with the final v0.23.0).

**Reply by pnmcosta** [reply]
Yup, confirmed that logs are also impacted by this issue, when I run with CGO_ENABLED=1 and only with it enabled, the `auxiliary.db`.`_logs` table doesn't get created, it's possibly a side effect of this issue rather than a new one?

Here's my findings, if it helps:

With or without a pre-existing logs.db file the second time I run the app I keep getting:

```
ERROR GET /api/logs?page=1&perPage=50&sort=-@rowid&skipTotal=1&filter=data.auth!="_superusers"
└─ Something went wrong while processing your request.
   └─ no such table: _logs; failed query: SELECT {{_logs}}.* FROM `_logs` WHERE JSON_EXTRACT([[data]], '$.auth') IS NOT {:tRa8LF} ORDER BY [[_logs]].[[_rowid_]] DESC LIMIT 50
```

and also when the app terminates:

```
2024/10/08 09:29:56 Failed to write log &{{ jguafmsvfj79kbj} 2024-10-08 08:29:51.726Z {"auth":"_superusers","execTime":43.483587,"method":"PATCH","referer":"http://127.0.0.1:8090/_/","remoteIP":"127.0.0.1","status":200,"type":"request","url":"/api/collections/g1jsonxecqx5egc/records/b2qaqf5c1kxl696","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36","userIP":"127.0.0.1"} PATCH /api/collections/g1jsonxecqx5egc/records/b2qaqf5c1kxl696 0} no such table: _logs
```

I manually checked the `auxiliary.db` created during the first time of running (that fails) and it is empty, without any tables.

This is not observed at all if running with CGO_ENABLED=0

**Reply by pnmcosta** [reply]
I don't think it's related to https://github.com/pocketbase/pocketbase/discussions/5618 as I observe this on the first run with CGO_ENABLED=1 (with or without a pre existing logs.db file), but as mentioned on my previous comment I'm inclined to think it's a side effect of the app crashing due to the unexpected erros caused by CGO driver columns cache issue.

**Reply by ganigeorgiev** [reply]
Honestly I don't know but I I don't have the time to further debug this right now.

As mentioned in the linked discussion, I'll consider eventually with the final v0.23.0 release to run a minimal set of required migrations for the auxiliary if the db or tables is missing but I need to cleanup other things first.

**Reply by pnmcosta** [reply]
Understandable, I personally wouldn't rate it high either.

Please don't judge me, I know you dislike getting asked for it, but I have to for syncing with my own feature releases, is there an ETA for final v0.23.0 release? 😅
